api/migrations/20181102040518-fix-old-image-urls-in-threads.js

```diff
@@ -0,0 +1,66 @@
+exports.up = async function(r, conn) {
+  const threads = await r
+    .db('spectrum')
+    .table('threads')
+    .filter(row => row('createdAt').lt(r.epochTime(1540929600)))
+    .filter(row => row('content')('body').match('spectrum.imgix.net'))
+    .filter(row => row('content')('body').match('%20'))
+    .filter(row => row.hasFields('deletedAt').not())
+    .map(row => ({ id: row('id'), body: row('content')('body') }))
+    .run(conn)
+    .then(cursor => cursor.toArray());
+
+  const threadPromises = threads.map(async obj => {
+    const newBody = JSON.parse(obj.body);
+
+    const imageKeys = Object.keys(newBody.entityMap).filter(
+      key => newBody.entityMap[key].type.toLowerCase() === 'image'
+    );
+
+    const LEGACY_PREFIX = 'https://spectrum.imgix.net/';
+    const hasLegacyPrefix = url => url.startsWith(LEGACY_PREFIX, 0);
+    const stripLegacyPrefix = url => url.replace(LEGACY_PREFIX, '');
+
+    const processImageUrl = str => {
+      if (str.indexOf(LEGACY_PREFIX) < 0) {
+        return str;
+      }
+      if (str.indexOf('%20') < 0) {
+        return str;
+      }
+
+      const split = str.split('?');
```

Comment:
This seems very finnicky, will this break images that have `?` in the filename? Is there a safer way to do this?

If this is a URL and you want to get the search, maybe we can parse the url with the Node core `import { URL } from 'url'` module?

Reply:
This works for now, let's run it. If this is too finnicky then we also need to edit how I've written the `editThread` and `publishThread` mutations :)